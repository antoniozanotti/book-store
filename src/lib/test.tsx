import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
