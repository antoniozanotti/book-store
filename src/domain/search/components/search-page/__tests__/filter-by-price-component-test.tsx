import { render } from "@/lib/test";
import { FilterByPriceComponent } from "../filter-by-price-component";
import { searchMock } from "@/domain/search/__tests__/search-mock";

jest.mock("../../../contexts/search-context", () => {
  return {
    useSearchContext: jest.fn(() => {
      return { state: searchMock };
    }),
  };
});

describe("FilterByPriceComponent", () => {
  it("should render", async () => {

    const { container } = render(<FilterByPriceComponent />);
    expect(container).toMatchSnapshot();
  });
});
