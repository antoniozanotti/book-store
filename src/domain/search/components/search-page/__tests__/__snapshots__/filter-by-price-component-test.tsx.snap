// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterByPriceComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <div
      class="flex flex-col gap-2 w-full h-full items-start justify-center overflow-hidden"
    >
      <div
        aria-labelledby="react-aria-:r1:"
        class="flex flex-col w-full gap-1"
        data-hover="false"
        data-orientation="horizontal"
        data-slot="base"
        id="react-aria-:r0:"
        role="group"
      >
        <div
          class="w-full flex justify-between items-center"
          data-slot="labelWrapper"
        >
          <label
            class="text-small"
            data-slot="label"
            id="react-aria-:r1:"
          >
            Price
          </label>
          <output
            aria-live="off"
            class="text-small"
            data-slot="value"
            for="react-aria-:r1:-0 react-aria-:r1:-1"
          >
            $20.00 â€“ $490.00
          </output>
        </div>
        <div
          class="relative flex gap-2 items-center"
          data-slot="track-wrapper"
        >
          <div
            class="flex w-full relative rounded-full bg-default-300/50 border-x-transparent h-1 my-[calc((theme(spacing.5)-theme(spacing.1))/2)] data-[thumb-hidden=false]:border-x-[calc(theme(spacing.5)/2)]"
            data-slot="track"
            data-thumb-hidden="false"
            data-vertical="false"
            style="position: relative;"
          >
            <div
              class="h-full absolute bg-primary"
              data-slot="filler"
              style="left: 0%; width: 98.12108559498957%;"
            />
            <div
              class="flex justify-center items-center before:absolute before:w-11 before:h-11 before:rounded-full after:shadow-small after:bg-background data-[focused=true]:z-10 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-5 h-5 after:w-4 after:h-4 rounded-full after:rounded-full bg-primary top-1/2 cursor-grab data-[dragging=true]:cursor-grabbing ring-transparent border-0 after:transition-all motion-reduce:after:transition-none data-[dragging=true]:after:scale-80 shadow-small"
              data-slot="thumb"
              index="0"
              orientation="horizontal"
              style="position: absolute; left: 0%; transform: translate(-50%, -50%);"
            >
              <div
                style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
              >
                <input
                  aria-describedby=""
                  aria-details=""
                  aria-labelledby="react-aria-:r1:"
                  aria-orientation="horizontal"
                  aria-valuetext="$20.00"
                  id="react-aria-:r1:-0"
                  max="490"
                  min="20"
                  step="10"
                  tabindex="0"
                  type="range"
                  value="20"
                />
              </div>
            </div>
            <div
              class="flex justify-center items-center before:absolute before:w-11 before:h-11 before:rounded-full after:shadow-small after:bg-background data-[focused=true]:z-10 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 w-5 h-5 after:w-4 after:h-4 rounded-full after:rounded-full bg-primary top-1/2 cursor-grab data-[dragging=true]:cursor-grabbing ring-transparent border-0 after:transition-all motion-reduce:after:transition-none data-[dragging=true]:after:scale-80 shadow-small"
              data-slot="thumb"
              index="1"
              orientation="horizontal"
              style="position: absolute; left: 98.12108559498957%; transform: translate(-50%, -50%);"
            >
              <div
                style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
              >
                <input
                  aria-describedby=""
                  aria-details=""
                  aria-labelledby="react-aria-:r1:"
                  aria-orientation="horizontal"
                  aria-valuetext="$490.00"
                  id="react-aria-:r1:-1"
                  max="499"
                  min="20"
                  step="10"
                  tabindex="0"
                  type="range"
                  value="490"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
